### ISSUE TITLE
[BLOCKER] Runtime Error in WP-Desktop: "TypeError: ...default is not a function..."

### ISSUE THREAD

**Issue by nsakaimbo** [Automattic/wp-calypso#40789]
Attempting to run WP Desktop with the latest Calypso master results in a runtime error: `TypeError: ...default is not a function...`. I tried debugging this (see my notes below) but haven't been successful and would appreciate some help.

#### Steps to reproduce

Update wp-desktop submodule with latest Calypso. Build and run the application.

#### Screenshots

<p align="center">
<img width="298" alt="desktop-runtime-error" src="https://user-images.githubusercontent.com/8979548/78412122-30331680-75e0-11ea-918f-e024bacf9f14.png">
</p>

#### Browser / OS version

macOS 10.15.3

#### Attempts to Debug

I tried to isolate the root cause but haven't had much luck. Perhaps these notes might be useful anyway.

Last known good SHA: 3af7284
First known bad SHA: e9cc8c0a687cd6f973ec607d460022ad41723c85

- I tried to perform a git bisect bisect between the aforementioned commits (with a full `npm ci` and rebuild at each step), but got inconsistent results (not sure why, I must be missing something)
- The error does not seem to occur in developer mode, only in production (where `NODE_ENV=production`). So perhaps webpack related?
- FWIW, the version of  Electron in the desktop app is running Node v10
- Some searching on the web indicates this might be related to the [inability to resolve export syntax](https://stackoverflow.com/questions/53200253/typeerror-default-is-not-a-function-import-export-javascript-class) somewhere in the code
- I also found [this discussion](https://github.com/iamkun/dayjs/issues/475#issuecomment-483792717) on Github regarding the use of `allowSyntheticDefaultImports` setting. I see we use this in the Calypso config - do we need to apply this in Desktop as well?

> Basically when you have a library compiling with tsconfig `allowSyntheticDefaultImports` and then a consuming application that does not use that setting, we got that error.

Not sure if any of the above notes are helpful but figured it wouldn't hurt to share.

**Comment by sirbrillig** [comment]
It's a long shot but this PR might be worth testing to see if it helps (by removing a test helper that imported a polyfill library): #40596

**Comment by nsakaimbo** [comment]
Thanks, @sirbrillig - I tried building a desktop branch with that PR, but unfortunately the runtime error still persists:

`https://github.com/Automattic/wp-desktop/tree/test-calypso-wip-pr`

Still a useful exercise in ruling out the imported polyfill library.

**Comment by nsakaimbo** [comment]
_Update_:

I believe I've been able to isolate the SHA that introduced the runtime error.

I tested a series of Calypso commits in a temporary branch `https://github.com/Automattic/wp-desktop/commits/test-calypso-runtime-error` in the desktop repo, and the error seems to have been introduced in SHA 8528342. I tested this SHA a couple of times to be sure, and also confirmed that the SHA immediately prior (d95191c) works as expected.

However nothing obvious jumps out to me with respect to the diff in 8528342, although I see that some webpack configuration items were changed.

@jsnajdr Wondering if you might have any educated guesses here?
